sdk/communication/AzureCommunicationCalling/ApiSamples.md

```diff
@@ -0,0 +1,760 @@
+self.call?.state
+
+// CallDirectionOutgoing = 1, CallDirectionIncoming = 2
+self.call?.callDirection
+
+// [Bool] isMicrophoneMuted - is local audio muted
+self.call?.isMicrophoneMuted
+
+// [Errror] callEndReason - containing code/subcode/message indicating how call ended
+self.call?.callEndReason
+
+// String[] localVideoStreams - collection of video streams send to other participants in a call
+self.call?.localVideoStreams
+
+// RemoteParticipant[] remoteParticipants - collection of remote participants participating in this call
+self.call?.remoteParticipants
+
+// Indicates if the call is currently being recorded or not
+self.call?.isRecordingActive;
+```
+
+### Call Operations
+* ### Make Outgoing Call
+
+To create and start a call you need to call one of the APIs on CallAgent
+```swift
+public class CallManager : NSObject, CallDelegate
+{
+    public func placeCall(names: [CommunicationIdentifier])
+    {
+        let callOptions = StartCallOptions()!
+        // Using the CallAgent received from the factory method
+        // and returns a Call object
+        let audioOptions = AudioOptions()
+        audioOptions!.muted = true // Start the call in mute state
+        callOptions.audioOptions = audioOptions
+        let call = self.callAgent?.startCall(participants: names,
+                                            options: callOptions)
+        call?.delegate = self.callObserver
+    }
+
+    public func placeVideoCall(names: [CommunicationIdentifier])
+    {
+        let camera = self.deviceManager?.cameras!.first
+        let localVideoStream = LocalVideoStream(camera: camera)
+        let videoOptions = VideoOptions(localVideoStream: localVideoStream)
+        
+        let callOptions = StartCallOptions()
+        callOptions?.videoOptions = videoOptions
+
+        let call = self.callAgent?.startCall(participants: names,
+                                                options: callOptions)
+        call?.delegate = self.callObserver
+    }
+
+    // Following is one event which is part of CallDelegate
+    public func OnStateChanged(_ call: Call!,
+                                args: PropertyChangedEventArgs!)
+    {
+        let state = CallObserver.callStateToString(state: call.state)
+        print("Call state changed to %@", state)
+    }
+}
+
+```
+
+* ### Accept Incoming Call
+
+To accept a call you need to call one of the APIs on CallAgent
+```swift
+public class CallManager : NSObject
+{
+    public func acceptCall()
+    {
+        let acceptCallOptions = AcceptCallOptions()!
+        let camera = self.deviceManager?.cameras!.first
+        let localVideoStream = LocalVideoStream(camera: camera)
+        let videoOptions = VideoOptions(localVideoStream: localVideoStream)
+        acceptCallOptions.videoOptions = videoOptions
+        self.incomingCall?.accept(options: acceptCallOptions) { (call, error) in
+            if(error == nil) {
```

Comment:
This is asynchronous call, need more info here.

Reply:
@tjprescott example ?

Reply:
```swift
 self.incomingCall?.accept(options: acceptCallOptions) { call, error in
  guard error == nil else {
    print("error")
    return
  }
  // otherwise it was successful
  ...
}
```

This would be the more idiomatic swift approach rather than if/else